tests/unit/modules/test_asset.py

```diff
@@ -13,25 +14,242 @@
+                        "url": "http://def.com",
+                        "sha512": "def",
+                        "filters": ["d == d", "e == e"],
+                        "headers": {
+                            "bar": "foo"
+                        }
+                    },
+
+                ]
+            },
+            {
+                "name": "asset",
+                "builds": [
+                    {
+                        "url": "http://def.com",
+                        "sha512": "def",
+                        "filters": ["e == e", "d == d"],
+                        "headers": {
+                            "bar": "foo"
+                        }
+                    },
+                    {
+                        "url": "http://abc.com",
+                        "sha512": "abc",
+                        "filters": ["b == b", "a == a"],
+                        "headers": {
+                            "foo": "bar"
+                        }
+                    },
+                ]
+            },
+        ) is False
+
+    def test_different_current_none(self):
+        assert asset.do_differ(
+            None,
+            {
+                "name": "a",
+                "builds": [
+                    {
+                        "url": "http://abc.com",
+                        "sha512": "abc",
+                    }
+                ]
+            },
+        ) is True
+
+    def test_different(self):
+        assert asset.do_differ(
+            {
+                "name": "asset",
+                "builds": [
+                    {
+                        "url": "http://abc.com",
+                        "sha512": "abc",
+                    }
+                ]
+            },
+            {
+                "name": "asset",
+                "builds": [
+                    {
+                        "url": "http://def.com",
+                        "sha512": "abc",
+                    },
+                    {
+                        "url": "http://def.com",
+                        "sha512": "abc",
+                        "filters": ["abc == def"],
+                    }
+                ]
+            },
+        ) is True
+
+
+class TestDoBuildsDiffer:
+    def test_equal_with_different_order(self):
+        assert asset._do_builds_differ(
+            [
+                {"sha512": "b", "url": "b"},
+                {"sha512": "a", "url": "a"},
+            ],
+            [
+                {"sha512": "a", "url": "a"},
+                {"sha512": "b", "url": "b"}
+            ],
+        ) is False
+
+    def test_equal_with_different_order_of_filters(self):
+        assert asset._do_builds_differ(
+            [{
+                "sha512": "a",
+                "url": "a",
+                "filters": ["a", "b", "c"],
+            }],
+            [{
+                "sha512": "a",
+                "url": "a",
+                "filters": ["c", "a", "b"],
+            }],
+        ) is False
```

Comment:
We already cover those two scenarios with the `TestDoDiffer.test_equal` test case. We can safely drop them.